src/domain/wallet-invoices/wallet-invoice-builder.ts

```diff
@@ -2,17 +2,37 @@ import { invoiceExpirationForCurrency } from "@domain/bitcoin/lightning"
 import { checkedToBtcPaymentAmount, checkedToUsdPaymentAmount } from "@domain/payments"
 import { WalletCurrency, ZERO_SATS } from "@domain/shared"
 
+import { InvalidWalletInvoiceBuilderStateError } from "./errors"
+
 export const WalletInvoiceBuilder = (
   config: WalletInvoiceBuilderConfig,
 ): WalletInvoiceBuilder => {
+  const withSecretAndHash = ({
```

Comment:
I'm not sure we need to expose this step - can the `getSecretAndPaymentHash` function be called from within the `WalletInvoiceBuilder`? Similar to how we call `LnFees` from the `PaymentFlowBuilder`? Its information that is generated within our app - not user input.
Getting rid of that step would remove some code - do you see a downside?

Reply:
Good point, made the change here: https://github.com/GaloyMoney/galoy/pull/1395/commits/641e7113456c1a51e57f65318f98c4f226c30e89